test/with-createreadstream-mockserver.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Google LLC
```

Comment:
nit - the test file is also a bit confusing. I think your intention is to differentiate the test between non-retryable error like NOT_FOUND v.s. a DEADLINE_EXCEEDED error. 

But the `createreadstream` v.s. `runquery` in the name made it confusing as it seems to do with the functionalities.

Reply:
The intent is to differentiate between the calls to the lookup grpc endpoint and the runQuery grpc endpoint.

Note that this PR is in a Draft stage. Although the early feedback is appreciated, I think I'm going to remove this file so that we can lock in the progress we have with the UNAVAILABLE test on runQuery. This gives us the confidence that we can see the error stack for these types of calls.